src/sentry/api/endpoints/organization_events_vitals.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import absolute_import
+import sentry_sdk
+
+from rest_framework.response import Response
+from rest_framework.exceptions import ParseError
+
+from sentry.api.bases import OrganizationEventsV2EndpointBase, NoProjects
+from sentry.api.event_search import get_function_alias
+from sentry.snuba import discover
+
+
+class OrganizationEventsVitalsEndpoint(OrganizationEventsV2EndpointBase):
+    VITALS = {
+        "measurements.lcp": {"thresholds": [0, 2500, 4000]},
+        "measurements.fid": {"thresholds": [0, 100, 300]},
+        "measurements.cls": {"thresholds": [0, 0.1, 0.25]},
+        "measurements.fcp": {"thresholds": [0, 1000, 3000]},
+        "measurements.fp": {"thresholds": [0, 1000, 3000]},
+    }
+
+    def get(self, request, organization):
+        if not self.has_feature(organization, request):
+            return Response(status=404)
+
+        with sentry_sdk.start_span(op="discover.endpoint", description="parse params"):
+            try:
+                params = self.get_snuba_params(request, organization)
+            except NoProjects:
+                return Response([])
+
+            vitals = [vital.lower() for vital in request.GET.getlist("vital", [])]
+            if len(vitals) == 0:
+                raise ParseError(detail=u"Need to pass at least one vital")
+
+            selected_columns = []
+            aliases = {}
+            for vital in vitals:
+                if vital not in self.VITALS:
+                    raise ParseError(detail=u"{} is not a valid vital".format(vital))
+                aliases[vital] = []
+                for index, threshold in enumerate(self.VITALS[vital]["thresholds"]):
+                    column = f"count_at_least({vital}, {threshold})"
+                    # Order aliases for later calculation
+                    aliases[vital].append(get_function_alias(column))
+                    selected_columns.append(column)
+                selected_columns.append(f"p75({vital})")
+
+        with self.handle_query_errors():
+            events_results = discover.query(
+                selected_columns=selected_columns,
+                query=request.GET.get("query"),
+                params=params,
+                # Results should only ever have 1 page
+                limit=1,
+                referrer="api.events.vitals",
+                auto_fields=True,
+                auto_aggregations=True,
+                use_aggregate_conditions=True,
+            )
+
+        results = {}
+        event_data = events_results["data"][0]
```

Comment:
Can we add a check to make sure there is exactly 1 row in the results? If there is a hiccup in snuba, we wouldn't want sentry to error too.